pkg/cstor/cvr_operation.go

```diff
@@ -30,14 +31,23 @@ var CVRWaitCount = 100
 // CVRCheckInterval defines amount of delay for CVR check
 var CVRCheckInterval = 5 * time.Second
 
-// waitForAllCVR will ensure that all CVR related to
-// given volumes are created
-func (p *Plugin) waitForAllCVR(vol *Volume) error {
+// waitForAllCVRs will ensure that all CVR related to
+// the given volume is created
+func (p *Plugin) waitForAllCVRs(vol *Volume) error {
 	replicaCount := p.getCVRCount(vol.volname)
 	if replicaCount == -1 {
 		return errors.Errorf("Failed to fetch replicaCount for volume{%s}", vol.volname)
 	}
 
+	if p.isCSIVolume {
+		return p.waitForCSIBasedCVRs(vol, replicaCount)
+	}
+	return p.waitFoNonCSIBasedCVRs(vol, replicaCount)
+}
+
+// waitFoNonCSIBasedCVRs will ensure that all CVRs related to
+// given non CSI based volume is created
+func (p *Plugin) waitFoNonCSIBasedCVRs(vol *Volume, replicaCount int) error {
```

Comment:
I see the only difference is version of the client.. can we just do a method that gives list of cvrs based on the version?
this is duplicating the code which doesn't seems to be necessary here

Reply:
ok.. problem might be the the return types of List API

Reply:
Yes, the types. Other improvements will require a good amount of refactoring and can be taken subsequently.